test/integration2/query/select/selects.spec.js

```diff
@@ -597,137 +598,116 @@ describe('Selects', function () {
           {
-            null_col: 'test',
-            string_col: 'a',
+            nulls: 'first',
+            order: 'asc',
+            expected: ['b', 'd', 'a', 'c'],
           },
           {
-            null_col: null,
-            string_col: 'b',
+            nulls: 'last',
+            order: 'asc',
+            expected: ['a', 'c', 'b', 'd'],
           },
           {
-            null_col: 'test2',
-            string_col: 'c',
+            nulls: 'first',
+            order: 'desc',
+            expected: ['b', 'd', 'c', 'a'],
           },
           {
-            null_col: null,
-            string_col: 'd',
+            nulls: 'last',
+            order: 'desc',
+            expected: ['c', 'a', 'b', 'd'],
           },
-        ]);
-
-        await knex('OrderByNullTest')
-          .pluck('id')
-          .orderBy([
-            { column: 'null_col', order: 'asc', nulls: 'first' },
-            { column: 'string_col' },
-          ])
-          .testSql(function (tester) {
-            tester(
-              'mysql',
-              'select `id` from `OrderByNullTest` order by (`null_col` is not null) asc, `string_col` asc',
-              [],
-              [2, 4, 1, 3]
-            );
-            tester(
-              'pg',
-              'select "id" from "OrderByNullTest" order by "null_col" asc nulls first, "string_col" asc',
-              [],
-              [2, 4, 1, 3]
-            );
-            tester(
-              'pgnative',
-              'select "id" from "OrderByNullTest" order by "null_col" asc nulls first, "string_col" asc',
-              [],
-              [2, 4, 1, 3]
-            );
-            tester(
-              'pg-redshift',
-              'select "id" from "OrderByNullTest" order by "null_col" asc nulls first, "string_col" asc',
-              [],
-              ['2', '4', '1', '3']
-            );
-            tester(
-              'sqlite3',
-              'select `id` from `OrderByNullTest` order by (`null_col` is not null) asc, `string_col` asc',
-              [],
-              [2, 4, 1, 3]
-            );
-            tester(
-              'oracledb',
-              'select "id" from "OrderByNullTest" order by "null_col" asc nulls first, "string_col" asc',
-              [],
-              [2, 4, 1, 3]
-            );
-            tester(
-              'mssql',
-              'select [id] from [OrderByNullTest] order by IIF([null_col] is null,0,1) asc, [string_col] asc',
-              [],
-              [2, 4, 1, 3]
-            );
-          });
+        ];
+
+        // test mixed strings and nulls, mixed ints and nulls
+        for (const type of ['string', 'int']) {
+          // test all combinations of null-order and value-order
+          const column = `nullable_${type}`;
+
+          for (const { nulls, order, expected } of table) {
+            it(`nulls ${nulls}, order ${order}, values ${type}, array-of-objects syntax`, async () => {
+              await assertRows(
```

Comment:
We already have unit tests on the SQL construction, so using `.testSql` was redundant here. It also passed when a specific dialect was not included in the `tester(...)` calls even if the query would fail on that dialect. The new integration tests simply run on whatever databases we're running the tests against, and expect all dialects to be able to satisfy the query